src/components/topNav/primaryTopNav/PrimaryTopNav.tsx

```diff
@@ -35,6 +36,20 @@ export const PrimaryTopNav = ({ items, mobileLogo, layoutClassName }: TopNavProp
     }
   };
 
+  const handleSubItemMenu = (idx: number, value: boolean) => {
+    setSubNavIsOpen({
+      ...subNavIsOpen,
+      [idx as keyof typeof subNavIsOpen]: value,
+    });
+  };
+
+  const screenWidth = window.innerWidth;
+
+  React.useEffect(() => {
+    if (screenWidth > 992) {
+      setSubNavIsOpen({});
+    }
+  }, [screenWidth]);
```

Comment:
I think this is not triggering on the "right" moments. `screenWidth` seems to update sometimes, I think whenever the React lifecycle is triggered by some other side effect.

To accurately keep track of a screen width, we can register (and un-register when needed) a listener, like so (dummy code):

```
const [screenWidth, setScreenWidth] = React.useState<number | undefined>(undefined);

  React.useEffect(() => {
    const handleScreenResize = (e: TypeThis) => setScreenWidth(e.path_to_width);

    window.addEventListener("resize", handleScreenResize); // add listener

    return () => window.removeEventListener("resize", handleScreenResize) // remove listener when component is removed
  }, []) // only trigger once
```